plugins/inventory/foreman.py

```diff
@@ -211,10 +219,14 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
         return hostvars
 
     def _populate(self):
+        host_filters = self.get_option('host_filters')
 
         for host in self._get_hosts():
-
             if host.get('name'):
+                # Skip any hosts that match host_filters
+                if host_filters:
+                    if any([re.search(host_filter, host['name']) for host_filter in host_filters]):
+                        continue
```

Comment:
the old script used the same config file option to pass a "search" to foreman:
https://github.com/ansible/ansible/blob/5192f68f3a056765bbdb6614aea478e4bd977335/contrib/inventory/foreman.py#L204-L206

using the same name but a completely different filtering seems odd to me. plus this will still fetch the host information from the API and discard it later, instead of not fetching it at all.

IMHO the change in #823 is cleaner and should be prefered.